### ISSUE TITLE
Delete all sessions by user

### ISSUE THREAD

**Issue by zwhitchcox** [boj/redistore#56]
So, I want to give my users the ability to sign out of all other devices. Is that something I could do with this software, or would I need to make my own solution?

**Comment by ptman** [comment]
If you don't want to give your users the ability to pick which sessions they want to sign out of, you could store a session token in the user data in the database. And only consider sessions that contain the session token to be valid. Then you can sign out all but the current session by changing the session token in the database and the current session. All the other sessions will no longer be valid.

**Comment by pssgo** [comment]
some issue

**Comment by appleboy** [comment]
redistore is not designed to solve the "sign out from all other devices" problem. This library serves as a session store backend for gorilla/sessions, providing Redis-based storage for individual session data. Its primary purpose is to manage the lifecycle of single sessions (create, read, update, delete) using Redis as the persistence layer.

The core limitation is that redistore operates at the session level, not the user level. Each session is stored independently in Redis with a randomly generated session ID (e.g., session_ABC123), with no built-in mechanism to associate multiple sessions with a single user. The library has no concept of "which sessions belong to user X" - it simply stores and retrieves session data by session ID.

To implement "sign out from all devices," you would need to build an additional layer on top of redistore that:
1. Maintains a user-to-sessions mapping (e.g., using Redis Sets: user_sessions:{userID})
2. Tracks all active session IDs for each user
3. Provides methods to invalidate multiple sessions for a given user

In summary, redistore gives you the building blocks (Redis connection pool, session serialization, secure cookie handling), but you'll need to implement your own solution for tracking and managing multiple sessions per user.

**Comment by ptman** [comment]
It's also possible to invalidate all older sessions by just having a per-user generation. Sessions have a generation, and if the user generation is increased all sessions with older generation are invalid.

Update: seems like I'm repeating myself from 5y ago.
